ratelimits/names.go

```diff
@@ -1,31 +1,42 @@
 package ratelimits
 
 import (
+	"fmt"
+	"net"
 	"strconv"
-	"strings"
 )
 
 // Name is an enumeration of all rate limit names. It is used to intern rate
 // limit names as strings and to provide a type-safe way to refer to rate
 // limits.
 //
 // IMPORTANT: If you add a new limit Name, you MUST add it to the 'nameToString'
-// mapping. Also, new IPv4Address and IPv6Range rate limit names MUST contain
-// the string "IPv4Address" and "IPv6Range", respectively.
+// mapping and idValidForName function below.
 type Name int
 
 const (
-	// UsageRequestsPerIPv4Address uses bucket key 'enum:ipv4address'.
+	// UsageRequestsPerIPv4Address uses bucket key 'enum:ipv4address'. Usage
+	// related paths are: new-nonce, new-account, new-order, or revoke-cert.
 	UsageRequestsPerIPv4Address Name = iota
 
-	// InfoRequestsPerIPv4Address uses bucket key 'enum:ipv4address'.
+	// UsageRequestsPerIPv6Range uses bucket key 'enum:ipv6rangeCIDR'. The
+	// address range must be a /48. Usage related paths are: new-nonce,
+	// new-account, new-order, or revoke-cert.
+	UsageRequestsPerIPv6Range
+
+	// InfoRequestsPerIPv4Address uses bucket key 'enum:ipv4address'. Info
+	// related paths are: directory and acme.
 	InfoRequestsPerIPv4Address
 
+	// InfoRequestsPerIPv6Range uses bucket key 'enum:ipv6rangeCIDR'. The
+	// address range must be a /48. Info related paths are: directory and acme.
+	InfoRequestsPerIPv6Range
```

Comment:
These limits, since they are reimplementations of nginx limits rather than existing boulder limits, feel like we're going to have a hard time comparing-and-contrasting their performance. My general feeling is that we shouldn't be adding rate limits to this enum until we're actually adding call-sites and configuration for them in individual boulder components. That feeling is especially true for these ones, which don't have existing analogues.

Also, it seems like it would be nice for us to build something _better_ than what nginx currently gives us. Like, what about key-change requests? ARI requests? Are we counting new-order in both limits because it's both a new-order and falls under the /acme path? It seems like maybe the thing we actually care about here is POST vs GET/POST-as-GET, in which case we could/should divide these things up a little differently.

Long story short, I'm not sure that these rate limit names and comments will remain the same as we go through the code review that involves actually instantiating them in the WFE, so I'm not sure that it's good to add them preemptively like this.

(The IPv4 and IPv6range validation code can still serve the NewRegistrations limit.)

Reply:
I've gone ahead and removed the nginx limits, and I'll constrain the scope of this PR to only the limits currently supported in boulder.